enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,854 @@
+  # Informing tests will not negatively impact default views
+  # in Component Readiness.
+  # However, policies may be established in the future that 
+  # cause tests that remain informing too long to be treated
+  # as if they are production.
+  "lifecycle": "informing",
+  
+}
+```
+
+##### Run-Test - Running Extension Tests
+The `run-test` verb is used to cause the extension binary to actually run 
+discovered tests. `run-test` accepts test environment arguments as discussed
+in the "list" verb section (in order to reduce fact discovery time needed
+for each test run) and one or more test names to invoke.
+```
+$ extension-binary run-test 
+  --component     "default" or "<product>:<type>:<component>"
+  --platform      The hardware or cloud platform ("aws", "gcp", "metal", ...).
+  ...other environment arguments...
+  --config        The component configuration profile to assume is active on the cluster.
+  --name | -n     Test name to invoke (-n can be specified multiple times).
+  --list          Filename or "--" for stdin of tests to invoke from 'list' output.
+```
+
+`run-test` should run the specified tests in parallel. To prevent 
+test cases developing a dependence on one another, `run-test` must make no
+assumption about the number or composition of tests it is asked to run
+(other than they will be consistent with the test environment and isolation
+requirements defined by the "list" verb).
+
+`openshift-tests` will randomize tests into multiple different parallel
+executions of the extension binary in order to amortize initialization
+costs.
+
+Standard output from the invocation will be serialized into JSONL with
+each line associated with a single test run result.
+
+```python
+{ "name": "T1", "result": "success", }
+{ "name": "T1", "result": "success", }
+{ "name": "T1", "result": "success", }
+```
+
+A single test result will contain the following information, encoded on a single line:
+```python
+{
+    "name": "test name",
+    
+    # RFC-3339 start and stop time in UTC with millisecond precision.
+    "startTime": "2026-01-02T15:04:05.000Z",
+    "endTime": "2026-01-02T15:04:06.840Z",
+    
+    # The outcome of the test; pass, fail, skip, timeout.
+    "result": "pass",
+    "output": "standard output of the test",
+    "error": "standard error of the test",
+    
+    # Human-readable messages to further explain 
+    # skips / timeouts / etc.
+    "details": [
+        "example message"
+    ]
+}
+```
+
+When developing an extension, it may be useful to be able to run a suite
+advertised by an extension directly. This can be accomplished using a
+combination of `list` and `run-test`.
+
+```
+$ extension list --suite my-suite --platform aws ... | extension run-test --platform aws ...  --list --
+```
+
+Note that `run-test` will blindly execute tests in the list as quickly as possible,
+in parallel, without consideration for system resources or parallelism constraints

```

Comment:
Just a note, OTE doesn't do parallel execution yet.  There's some oddities about how we invoke ginkgo tests today.  I think I just haven't found all the things I need mutexes for yet.

Reply:
You can resolve this one, OTE executes in parallel.  Ginkgo has a mutex to force serial execution but other frameworks would be parallelized.